### PR TITLE
add in a profile api (fixes #950)

### LINKED ISSUE #950: Provide a user end point
This is for: https://bugzilla.mozilla.org/show_bug.cgi?id=1214789

At some point we'll roll out a user profile page. I think this is a great example end point. So let's do a user profile end point, maybe similar to http://firefox-marketplace-api.readthedocs.org/en/latest/topics/accounts.html#get--api-v2-account-settings-mine-

### PR DESCRIPTION
**Pull request by andymckay**
* create a profile end point
* refactor out the API auth base
* fixes #950

### CODE REVIEW

File: apps/accounts/tests/test_views.py [hunk 1]
```diff
@@ -64,3 +67,23 @@ def test_identify_success_account_exists(self):
...
+    def setUp(self):
+        self.create_api_user()
+        self.url = reverse('accounts.profile')
+        self.cls = resolve(self.url).func.cls
+        super(TestProfileView, self).setUp()
+
+    def test_good(self):
+        res = self.get(self.url)
+        assert res.status_code == 200
+        assert json.loads(res.content)['email'] == 'a@m.o'
```
Comment (mstriemer) [inline apps/accounts/tests/test_views.py]: You should be able to use `res.data`.

File: apps/accounts/views.py [hunk 2]
```diff
@@ -28,3 +31,12 @@ def post(self, request):
...
             login(request._request, user)
             return Response({'email': identity['email']})
+
+
+class ProfileView(JWTProtectedView, generics.RetrieveAPIView):
+    serializer_class = UserProfileSerializer
+
+    def retrieve(self, request, *args, **kw):
+        instance = request.user
+        serializer = self.get_serializer(instance)
```
Comment (mstriemer) [inline apps/accounts/views.py]: Inlining `request.user` looks like it would be fine here.

File: apps/api/tests/utils.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        to do a full request response cycle.
+        """
+        assert cls.authentication_classes == [JWTKeyAuthentication]
+
+    def verbs_allowed(self, cls, verbs):
+        """
+        Tests that the verbs you expect on the class are present and no more.
+        Options is added if you don't pass it.
+        """
+        verbs = set([v.upper() for v in verbs])
```
Comment (mstriemer) [inline apps/api/tests/utils.py]: The `[]` shouldn't be required. I'm not sure if there's a benefit one way or the other though.

File: apps/accounts/serializers.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = (
+            'username', 'display_name', 'email',
+            'bio', 'deleted', 'display_collections',
+            'display_collections_fav', 'homepage',
+            'location', 'notes', 'notifycompat',
+            'notifyevents', 'occupation', 'picture_type',
```
Comment (mstriemer) [inline apps/accounts/serializers.py]: Can you include the URL for the picture?

I feel like the list of fields could be trimmed down too. We shouldn't need `failed_login_attempts`, `last_login_attempt`, `last_login_attempt_ip` and `t_shirt_requested` (is that still active?). `notifyevents` and `notifycompat` can probably go away too but I don't know what they're for. They should at least get named `notify_events` or break out into another object like `notify: {events: true, compat: false}`.

**Comment by mstriemer** [comment]
r+wc

File: apps/accounts/serializers.py (hunk 4 above)
Comment (andymckay) [inline apps/accounts/serializers.py]: I don't know what a lot of these fields are, wonder if @magopian does?

We can certainly expand these out if we need to and its probably easy to push something out with minimal fields.

File: apps/api/tests/utils.py (hunk 3 above)
Comment (andymckay) [inline apps/api/tests/utils.py]: Will remove.

File: apps/accounts/views.py (hunk 2 above)
Comment (andymckay) [inline apps/accounts/views.py]: good point

File: apps/accounts/serializers.py (hunk 4 above)
Comment (andymckay) [inline apps/accounts/serializers.py]: Although `last_login_* stuff I can see being useful to a user and something that will be shown on the profile page: https://addons.mozilla.org/en-US/firefox/user/amckay/

Comment (mstriemer) [inline apps/accounts/serializers.py]: You could leave it. I'm not sure how we'd track login failure once we land FxA though.

Comment (andymckay-limited-access) [inline apps/accounts/serializers.py]: Good point! Removed.

**Pull request merged**